### PR TITLE
Fix verify install kinds for kind NetworkAttachmentDefinition

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Fix https://github.com/istio/istio/issues/46859

### CODE REVIEW

**Comment by hanxiaop** [comment]
Verified it's working in my environment:
```
 istioctl verify-install                          
1 Istio control planes detected, checking --revision "default" only
...
✔ DaemonSet: istio-cni-node.istio-system checked successfully
✔ NetworkAttachmentDefinition: istio-cni.default checked successfully  # this one
✔ ServiceAccount: istio-cni.istio-system checked successfully
✔ DaemonSet: ztunnel.istio-system checked successfully
✔ ServiceAccount: ztunnel.istio-system checked successfully
✔ ServiceAccount: istio-reader-service-account.istio-system ch
```

File: istioctl/pkg/verifier/verifier.go [hunk 1]
```diff
@@ -414,6 +411,18 @@ func (v *StatusVerifier) verifyPostInstall(visitor resource.Visitor, filename st
 	return crdCount, istioDeploymentCount, daemonSetCount, err
 }
 
+func resourceKinds(un *unstructured.Unstructured) string {
+	kinds := findResourceInSpec(un.GetObjectKind().GroupVersionKind())
+	if kinds == "" {
+		kinds = strings.ToLower(un.GetKind()) + "s"
+	}
+	// Fix the specific kind which does not follow the general pattern.
+	if un.GetKind() == "NetworkAttachmentDefinition" {
```
Comment (zirain) [inline istioctl/pkg/verifier/verifier.go:420]: use a map the future?

**Review by jwendell — APPROVED** [review]

File: istioctl/pkg/verifier/verifier.go (hunk 1 above)
Comment (hanxiaop) [reply istioctl/pkg/verifier/verifier.go]: Fixed, thanks

File: istioctl/pkg/verifier/verifier.go [hunk 2]
```diff
@@ -414,6 +411,22 @@ func (v *StatusVerifier) verifyPostInstall(visitor resource.Visitor, filename st
 	return crdCount, istioDeploymentCount, daemonSetCount, err
 }
 
+func resourceKinds(un *unstructured.Unstructured) string {
+	kinds := findResourceInSpec(un.GetObjectKind().GroupVersionKind())
+	if kinds == "" {
+		kinds = strings.ToLower(un.GetKind()) + "s"
+	}
+
+	specialKinds := map[string]string{
```
Comment (zirain) [inline istioctl/pkg/verifier/verifier.go:420]: move out of the function?

Comment (hanxiaop) [reply istioctl/pkg/verifier/verifier.go]: Done, thanks

**Review by GregHanson — APPROVED** [review]

**Pull request merged**